### PR TITLE
chore: [CO-410] maven migration

### PR DESCRIPTION
**Pull request by frisonisland**
This PR migrates the repo from ant + ivy to maven.
The profile "dev" has been added to produce -SNAPSHOT if needed.

Changes have been picked up from another branch. I've decided to split the maven refactor in this PR (without feature of course) to make changes clearer.

### CODE REVIEW

**Review by keshavbhatt — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,3 +1,7 @@
+def mvnCmd(String cmd) {
+  sh 'mvn -s settings.xml ' + cmd
```
Comment (Polpetta) [inline Jenkinsfile]: Add `-B` to avoid spamming in the logs

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      <id>liferay</id>
+      <url>https://repository.liferay.com/nexus/content/repositories/public/</url>
+    </repository>
+  </repositories>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.8.2</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>4.4.0</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
+      <version>3.1.0</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.dom4j</groupId>
+      <artifactId>dom4j</artifactId>
+      <version>2.1.1</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>28.1-jre</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
+      <version>4.5.13</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpcore</artifactId>
+      <version>4.4.11</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>commons-io</groupId>
+      <artifactId>commons-io</artifactId>
+      <version>1.4</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+    <dependency>
+      <groupId>commons-lang</groupId>
+      <artifactId>commons-lang</artifactId>
+      <version>2.6</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+    </dependency>
+  </dependencies>
+
+  <build>
+      <plugins>
+        <!-- MANIFEST properties for Extension  class-->
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-jar-plugin</artifactId>
+          <version>3.3.0</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+                <Zimbra-Extension-Class>com.zimbra.cert.ZimbraCertMgrExt</Zimbra-Extension-Class>
+              </manifestEntries>
+            </archive>
+          </configuration>
+        </plugin>
+        <!-- Testing -->
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>2.22.2</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+        </plugin>
+        <!-- Code Coverage -->
+        <plugin>
+          <groupId>org.jacoco</groupId>
+          <artifactId>jacoco-maven-plugin</artifactId>
+          <version>0.8.8</version>
```
Comment (Polpetta) [inline pom.xml]: Move it as properties please

**Review by Polpetta — CHANGES_REQUESTED** [review]
There are some minor refirements I've left you to check, the rest is pretty much ok! :+1:

**Comment by frisonisland** [comment]
> There are some minor refirements I've left you to check, the rest is pretty much ok! +1

Thanks, I moved everything to properties

**Review by Polpetta — APPROVED** [review]

**Pull request merged**